sky/global_user_state.py

```diff
@@ -413,171 +441,172 @@ def _get_user_hash_or_current_user(user_hash: Optional[str]) -> str:
-        return int(launch_time)
-    return None
+        return int(row.launched_at)
 
 
 def get_cluster_info(cluster_name: str) -> Optional[Dict[str, Any]]:
-    rows = _DB.cursor.execute('SELECT metadata FROM clusters WHERE name=(?)',
-                              (cluster_name,))
-    for (metadata,) in rows:
-        if metadata is None:
+    with Session(_SQLALCHEMY_ENGINE) as session:
+        row = session.query(cluster_table).filter_by(name=cluster_name).first()
+        if row is None or row.metadata is None:
             return None
-        return json.loads(metadata)
-    return None
+        return json.loads(row.metadata)
 
 
 def set_cluster_info(cluster_name: str, metadata: Dict[str, Any]) -> None:
-    _DB.cursor.execute('UPDATE clusters SET metadata=(?) WHERE name=(?)', (
-        json.dumps(metadata),
-        cluster_name,
-    ))
-    count = _DB.cursor.rowcount
-    _DB.conn.commit()
-    assert count <= 1, count
-    if count == 0:
-        raise ValueError(f'Cluster {cluster_name} not found.')
+    with Session(_SQLALCHEMY_ENGINE) as session:
+        count = session.query(cluster_table).filter_by(
+            name=cluster_name).update(
+                {cluster_table.c.metadata: json.dumps(metadata)})
+        session.commit()
+        assert count <= 1, count
+        if count == 0:
+            raise ValueError(f'Cluster {cluster_name} not found.')
 
 
 def get_cluster_storage_mounts_metadata(
         cluster_name: str) -> Optional[Dict[str, Any]]:
-    rows = _DB.cursor.execute(
-        'SELECT storage_mounts_metadata FROM clusters WHERE name=(?)',
-        (cluster_name,))
-    for (storage_mounts_metadata,) in rows:
-        if storage_mounts_metadata is None:
+    with Session(_SQLALCHEMY_ENGINE) as session:
+        row = session.query(cluster_table).filter_by(name=cluster_name).first()
+        if row is None or row.storage_mounts_metadata is None:
             return None
-        return pickle.loads(storage_mounts_metadata)
-    return None
+        return pickle.loads(row.storage_mounts_metadata)
 
 
 def set_cluster_storage_mounts_metadata(
         cluster_name: str, storage_mounts_metadata: Dict[str, Any]) -> None:
-    _DB.cursor.execute(
-        'UPDATE clusters SET storage_mounts_metadata=(?) WHERE name=(?)', (
-            pickle.dumps(storage_mounts_metadata),
-            cluster_name,
-        ))
-    count = _DB.cursor.rowcount
-    _DB.conn.commit()
-    assert count <= 1, count
-    if count == 0:
-        raise ValueError(f'Cluster {cluster_name} not found.')
+    with Session(_SQLALCHEMY_ENGINE) as session:
+        count = session.query(cluster_table).filter_by(
+            name=cluster_name).update({
+                cluster_table.c.storage_mounts_metadata:
+                    pickle.dumps(storage_mounts_metadata)
+            })
+        session.commit()
+        assert count <= 1, count
+        if count == 0:
+            raise ValueError(f'Cluster {cluster_name} not found.')
```

Comment:
just wondering for these kind of handling after the session commit. Do they have to within the `context`? Similar to other places where we have the rows to be processed within the session context. Wondering if they need to be taken out of the context.